darwin/future/core/types/common.py

```diff
@@ -0,0 +1,39 @@
+from typing import Dict
+
+from darwin.future.data_objects import validators as darwin_validators
+
+
+class TeamSlug(str):
+    """Team slug type"""
+
+    min_length = 1
+    max_length = 256
+
+    @classmethod
+    def __get_validators__(cls):  # type: ignore
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v: str) -> "TeamSlug":
+        assert len(v) < cls.max_length, f"maximum length for team slug is {cls.max_length}"
+        assert len(v) > cls.min_length, f"minimum length for team slug is {cls.min_length}"
+        if not isinstance(v, str):
+            raise TypeError("string required")
+        modified_value = darwin_validators.parse_name(v)
+        return cls(modified_value)
+
+    def __repr__(self) -> str:
+        return f"TeamSlug({super().__repr__()})"
+
+
+class QueryString:
```

Comment:
Kind of a much of a muchness with this, but thoughts on making this a basemodel and using a validtor instead? To keep it in line with our other code format

```python
class QueryString(BaseModel):
    value: Dict[str, str] # will provide mypy type hinting here and validation, but not ValidationError on passing a Dict[Any, Any]
    @pydantic.validator('value')
    def dict_check(cls, v) -> Dict[str, str]:
        assert all(isinstance(k, str) and isinstance(v, str) for k, v in value.items())
        return v

Reply:
Yeah, I like it